api/services/SiteCreator.js

```diff
@@ -82,6 +82,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
 const createSiteFromTemplate = ({ siteParams, user, template }) => {
   let site = Site.build(siteParams)
   site.engine = "jekyll"
+  site.defaultBranch = templateForTemplateName(template).branch
```

Comment:
The template object is loaded based on the template name which is what comes from the frontend. If it's unable to load a template, it'll throw an error before it gets here.